lib/web3-contracts.js

```diff
@@ -40,83 +45,53 @@ export function useKnownContract(name, signer = true) {
 }
 
 export function useTokenDecimals(symbol) {
-  const [decimals, setDecimals] = useState(-1)
-  const tokenContract = useKnownContract(`TOKEN_${symbol}`)
-
-  useEffect(() => {
-    let cancelled = false
-    setDecimals(-1)
-
-    if (symbol === 'ETH') {
-      setDecimals(18)
-    } else if (tokenContract) {
-      tokenContract.decimals().then(decimals => {
-        if (!cancelled) {
-          setDecimals(decimals)
-        }
-      })
-    }
-
-    return () => (cancelled = true)
-  }, [symbol, tokenContract])
-
-  return decimals
+  return tokenDecimals.get(symbol)
 }
 
-export function useEthBalance() {
-  const { account, web3ReactContext } = useWeb3Connect()
-  const [balance, setBalance] = useState(bigNum(-1))
+export function useAnjTotalSupply() {
```

Comment:
I was thinking of just inlining this call on `useBondingCurvePrice`. What do you think?

Reply:
Yes, I think we should just inline this there for now to simplify. We can always pull it out to be an argument in the future if we'd like to.